internal/controller/managedcluster_controller.go

```diff
@@ -107,36 +107,30 @@ func (r *ManagedClusterReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 	return r.Update(ctx, managedCluster)
 }
 
-func (r *ManagedClusterReconciler) setStatusFromClusterStatus(ctx context.Context, managedCluster *hmc.ManagedCluster) (requeue bool, _ error) {
+func (r *ManagedClusterReconciler) setStatusFromChildObjects(ctx context.Context, managedCluster *hmc.ManagedCluster, resourceID schema.GroupVersionResource) (requeue bool, _ error) {
 	l := ctrl.LoggerFrom(ctx)
 
-	resourceID := schema.GroupVersionResource{
-		Group:    "cluster.x-k8s.io",
-		Version:  "v1beta1",
-		Resource: "clusters",
-	}
-
 	list, err := r.DynamicClient.Resource(resourceID).Namespace(managedCluster.Namespace).List(ctx, metav1.ListOptions{
 		LabelSelector: labels.SelectorFromSet(map[string]string{hmc.FluxHelmChartNameKey: managedCluster.Name}).String(),
 	})
 
 	if apierrors.IsNotFound(err) || len(list.Items) == 0 {
-		l.Info("Clusters not found, ignoring since object must be deleted or not yet created")
+		l.Info(fmt.Sprintf("%s not found, ignoring since object must be deleted or not yet created", resourceID.Resource))
 		return true, nil
 	}
 
 	if err != nil {
-		return true, fmt.Errorf("failed to get cluster information for managedCluster %s in namespace: %s: %w",
-			managedCluster.Namespace, managedCluster.Name, err)
+		return true, fmt.Errorf("failed to get %s information for managedCluster %s in namespace: %s: %w",
+			resourceID.Resource, managedCluster.Namespace, managedCluster.Name, err)
 	}
 	conditions, found, err := unstructured.NestedSlice(list.Items[0].Object, "status", "conditions")
```

Comment:
in general case it could be 0 or multiple MachineDeployment objects in a template

Reply:
Line 117 checks for `len(list.Items) == 0`

Reply:
And it does requeue every 10 seconds even if there's no MachineDeployment object at all which is incorrect. Also multiple machinedeployments are not covered.